packages/cli/src/commands/setup/ui/tasks/setup-component-library.js

```diff
@@ -0,0 +1,203 @@
+import fs from 'fs'
+
+import { getPaths } from '../../../../lib'
+
+/**
+ * @param {string} str The value to search for.
+ * @returns true if the contents of App.[js|tsx] contains the given string, false otherwise.
+ */
+export function appJSContains(str) {
+  const webAppPath = getPaths().web.app
+  const content = fs.readFileSync(webAppPath).toString()
+  return content.includes(str)
+}
+
+/**
+ * Inject code into App.[js|tsx].
+ * Use of insertComponent assumes only one of (around|within) are used, and that the component
+ * identified by (around|within) occurs exactly once in App.js|tsx.
+ * Imports are added after the last redwoodjs import.
+ * moduleScopeLines are added after the last import.
+ *
+ * @param {Object} options Configure behavior
+ * @param {Object} options.insertComponent Configure component-inserting behavior
+ * @param {Object} options.insertComponent.name Name of component to insert.
+ * @param {Object|string} options.insertComponent.props Properties to pass to the inserted component.
+ * @param {string} options.insertComponent.around Name of the component around which the new
+ * component will be inserted. Mutually exclusive with insertComponent.within.
+ * @param {string} options.insertComponent.within Name of the component within which the new
+ * component will be inserted. Mutually exclusive with insertComponent.around.
+ * @param {string} options.insertComponent.insertBefore Content to insert before the inserted
+ * component.
+ * @param {string} options.insertComponent.insertAfter Content to insert after the inserted
+ * component.
+ * @param {Array} options.imports Import declarations to inject after the last redwoodjs import.
+ * @param {Array} options.moduleScopeLines Lines of code to inject after the last import statement.
+ * @returns Nothing; writes changes directly into App.js|tsx.
+ */
+export function extendAppJS({
+  insertComponent: {
+    name = undefined,
+    props = undefined,
+    around = undefined,
+    within = undefined,
+    insertBefore = undefined,
+    insertAfter = undefined,
+  },
+  imports = [],
+  moduleScopeLines = [],
+}) {
+  const webAppPath = getPaths().web.app
+  let content = fs.readFileSync(webAppPath).toString().split('\n')
+
+  if (moduleScopeLines && moduleScopeLines.length) {
+    content.splice(
+      content.findLastIndex((l) => l.trimStart().startsWith('import')) + 1,
+      0,
+      '', // Empty string to add a newline when we .join('\n') below.
+      ...moduleScopeLines
+    )
+  }
+
+  if (imports && imports.length) {
+    content.splice(
+      content.findLastIndex((l) => l.includes('@redwoodjs')) + 1,
+      0,
+      '', // Empty string to add a newline when we .join('\n') below.
+      ...imports
+    )
+  }
+
+  if (name) {
+    if ((around && within) || !(around || within)) {
+      throw new Error(
+        'To insert a component, exactly one of (around | within) must be defined. Choose one.'
+      )
+    }
```

Comment:
This isn't needed here, right? As this check is carried out inside of `insertComponent`

Reply:
Could have sworn I deleted that one. Thanks, I’ll pull it.